### DISCUSSION TITLE
Isolated virtual envs.

### DISCUSSION THREAD

**Discussion by SekouDiaoNlp** [juancarlospaco/plz#6] [Ideas]
Hi @juancarlospaco how are you?

Are you still working on plz? If that's still the case are there some features you would like help with implementing? Especially pertaining to creating/managing virtual envs à la python venv pipx/pipenv.

I played a bit with plz and find it pretty nice.

Let me know if you would be interested.

Cheers.

**Comment by juancarlospaco** [comment]
Hi, welcome back! I was thinking you were gone...,
you have admin permissions already on this repo, feel free to implement whatever you want,
just send pull requests, no problem.

**Reply by SekouDiaoNlp** [reply]
Hi, know I unfortunately had a lot of health issues lately but I am back. I fixed the issues with n'importer already. So now I was thinking about contributing to plz.

Do you have features you would like to have implemented but not have time to implement, or some other features that current `plz` user have asked you to implement?

**Comment by SekouDiaoNlp** [comment]
One great feature of `pipx` is the ability to isolate each installed application from the others and be able to call them directly from the command line without cluttering the user's environment.
Also being able to install and manage multiple versions of the same software without them interfering with each other is pretty awesome.

Starting by implementing a database of the hashes of each version of each software/library would be a good start.

**Comment by juancarlospaco** [comment]
@SekouDiaoNlp
If you want, these are some ideas, some are just done, but needs more manual testing, to improve the UX

- Upload a package to PYPI with the `plz upload` feature. This should work AFAIK.
- [Improve the CI build YAML](https://github.com/juancarlospaco/plz/blob/master/.github/workflows/main.yml) ?, maybe just copy the one from the cpython repo.

**Reply by SekouDiaoNlp** [reply]
I will have a look at it.

Have a good day
